src/Dosar.elm

```diff
@@ -28,15 +28,30 @@ empty =
         }
 
 
-view : Dosar -> (Dosar -> Cmd msg -> Sub msg -> msg) -> Html msg
+type Msg
+    = SetTemei Temei
+    | SetDocumentExecutoriu DocumentExecutoriu
+    | SetActiune Actiune
+
+
+update : Msg -> Model -> Model
+update msg (Dosar data) =
+    case msg of
+        SetTemei v ->
+            Dosar { data | temei = v }
+
+        SetDocumentExecutoriu v ->
+            Dosar { data | documentExecutoriu = v }
+
+        SetActiune v ->
+            Dosar { data | actiune = v }
+
+
+view : Model -> (Msg -> msg) -> Html msg
 view (Dosar data) callback =
-    let
-        c data =
-            callback (Dosar data)
-    in
-        div []
-            [ h1 [] [ text "Dosar nou" ]
-            , Temei.view data.temei (\v -> c { data | temei = v })
-            , DocumentExecutoriu.view data.documentExecutoriu (\v -> c { data | documentExecutoriu = v } Cmd.none Sub.none)
-            , Actiune.view data.actiune (\v -> c { data | actiune = v })
-            ]
+    div []
+        [ h1 [] [ text "Dosar nou" ]
+        , Temei.view data.temei (\v cmd sub -> callback (SetTemei v))
+        , DocumentExecutoriu.view data.documentExecutoriu (callback << SetDocumentExecutoriu)
```

Comment:
This line `55` embodies (I think) the approach you suggested. (The other 2 — `54` and `56` — are not there yet.)